rotkehlchen/tests/unit/decoders/test_cowswap.py

```diff
@@ -0,0 +1,366 @@
+        HistoryBaseEntry(
+            event_identifier=evmhash,
+            sequence_index=0,
+            timestamp=TimestampMS(1676987159000),
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal('0.001768460133875456')),
+            location_label=user_address,
+            notes='Burned 0.001768460133875456 ETH for gas',
+            counterparty=CPT_GAS,
+        ), HistoryBaseEntry(
+            event_identifier=evmhash,
+            sequence_index=1,
+            timestamp=TimestampMS(1676987159000),
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.DEPOSIT,
+            event_subtype=HistoryEventSubType.PLACE_ORDER,
+            asset=A_ETH,
+            balance=Balance(amount=FVal('24.311042505395616962')),
+            location_label=user_address,
+            notes='Deposit 24.311042505395616962 ETH to swap it for USDC in cowswap',
+            counterparty=CPT_COWSWAP,
+        ),
+    ]
+    assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('ethereum_accounts', [['0xdc4CaDC65123Ebd371887CaD59Cc8c6F8F6fC29c']])
+def test_invalidate_eth_order(database, ethereum_inquirer, ethereum_accounts):
+    tx_hex = deserialize_evm_tx_hash('0x5769b4634ae26ec93aebc80a50e0676b0793af485041b249652bd7ee6703a9f5')  # noqa: E501
+    evmhash = deserialize_evm_tx_hash(tx_hex)
+    user_address = ethereum_accounts[0]
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hex,
+    )
+    expected_events = [
+        HistoryBaseEntry(
+            event_identifier=evmhash,
+            sequence_index=0,
+            timestamp=TimestampMS(1677040511000),
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal('0.001171136978414093')),
+            location_label=user_address,
+            notes='Burned 0.001171136978414093 ETH for gas',
+            counterparty=CPT_GAS,
+        ), HistoryBaseEntry(
+            event_identifier=evmhash,
+            sequence_index=1,
+            timestamp=TimestampMS(1677040511000),
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.WITHDRAWAL,
+            event_subtype=HistoryEventSubType.CANCEL_ORDER,
+            asset=A_ETH,
+            balance=Balance(amount=FVal('50')),
+            location_label=user_address,
+            notes='Invalidate an order that intended to swap 50 ETH in cowswap',
+            counterparty=CPT_COWSWAP,
+        ),
+    ]
+    assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('ethereum_accounts', [[
+    '0x0214aE5fD178986fA18ff792e0b995Dc6a78cD56',
```

Comment:
same here. 

You have to make all these tests from the side ofthe user. Who is the user of cowswap here? The guys who calls the contract to do the refund? Or the guy who receives it? From my understanding it's the guy who receives it.

Just tracking all possible addresses is not the right way to write tests.

Reply:
Okay, makes sense